# 16-02-2017: Abdul Moiz <abdul.moiz@nxp.com>
# Changes done for blinking Power LED until device is booted fully
diff -Naur ./arch/arm/mach-comcerto/board-c2krv16x-26x.c ../../b/linux-4.1.8/arch/arm/mach-comcerto/board-c2krv16x-26x.c
--- ./arch/arm/mach-comcerto/board-c2krv16x-26x.c	2017-02-11 00:12:06.326896185 +0530
+++ ../../b/linux-4.1.8/arch/arm/mach-comcerto/board-c2krv16x-26x.c	2017-02-13 17:57:34.750891032 +0530
@@ -82,6 +82,7 @@
 extern void device_init(void);
 extern void mac_addr_init(struct comcerto_pfe_platform_data *);
 extern struct sys_timer comcerto_timer;
+extern void power_led_blink(void);
 
 static void __init board_gpio_init(void)
 {
@@ -864,6 +865,8 @@
 	device_init();
 	board_gpio_init();
 
+	power_led_blink();
+
 #if defined(CONFIG_COMCERTO_I2C_SUPPORT)
 	i2c_register_board_info(0, comcerto_i2c_board_info, ARRAY_SIZE(comcerto_i2c_board_info));
 #endif
diff -Naur ./arch/arm/mach-comcerto/Makefile ../../b/linux-4.1.8/arch/arm/mach-comcerto/Makefile
--- ./arch/arm/mach-comcerto/Makefile	2017-02-11 00:12:06.354896186 +0530
+++ ../../b/linux-4.1.8/arch/arm/mach-comcerto/Makefile	2017-02-13 17:46:17.042868144 +0530
@@ -24,7 +24,8 @@
 obj-$(CONFIG_RTSM_C2K) 				+= board-c2krtsm.o
 obj-$(CONFIG_C2K_EVM) 				+= board-c2kevm.o
 obj-$(CONFIG_C2K_MFCN_EVM)			+= board-c2kmfcnevm.o
-obj-$(CONFIG_C2K_RV16X_26X)			+= board-c2krv16x-26x.o
+obj-$(CONFIG_C2K_RV16X_26X)			+= board-c2krv16x-26x.o \
+						   power_led_c2krv16x-26x.o
 obj-$(CONFIG_C2K_ASIC) 				+= board-c2kasic.o
 obj-$(CONFIG_SMP)				+= platsmp.o headsmp.o
 obj-$(CONFIG_COMCERTO_MSP)			+= msp/
diff -Naur ./arch/arm/mach-comcerto/power_led_c2krv16x-26x.c ../../b/linux-4.1.8/arch/arm/mach-comcerto/power_led_c2krv16x-26x.c
--- ./arch/arm/mach-comcerto/power_led_c2krv16x-26x.c	1970-01-01 05:30:00.000000000 +0530
+++ ../../b/linux-4.1.8/arch/arm/mach-comcerto/power_led_c2krv16x-26x.c	2017-02-13 17:47:17.162870174 +0530
@@ -0,0 +1,66 @@
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/kernel.h>
+#include <linux/gpio.h>
+#include <linux/interrupt.h>
+#include <linux/io.h>
+#include <linux/delay.h>
+#include <linux/kthread.h>
+#include <linux/proc_fs.h>
+#include <linux/seq_file.h>
+#include <asm/uaccess.h>
+
+static struct task_struct *blink_thread;
+int full_boot = 0;
+char msg;
+
+int power_led_blink_thread(void);
+void power_led_blink(void);
+
+int power_led_proc_write(struct file *filp,const char *buf,size_t count,loff_t *offp)
+{
+	copy_from_user(&msg,buf,1);
+
+	if(msg == '1')
+		full_boot = 1;
+	
+	return count;
+}
+
+struct file_operations power_led_fops = {
+	write: power_led_proc_write
+};
+
+int power_led_blink_thread()
+{
+	writel(readl(COMCERTO_GPIO_63_32_PIN_SELECT) | GPIO_PIN_59, COMCERTO_GPIO_63_32_PIN_SELECT);
+	writel(readl(COMCERTO_GPIO_63_32_PIN_OUTPUT_EN) & ~GPIO_PIN_59, COMCERTO_GPIO_63_32_PIN_OUTPUT_EN);
+
+	while(!full_boot)
+	{
+		//POWER_LED OFF
+		writel(readl(COMCERTO_GPIO_63_32_PIN_OUTPUT) & ~GPIO_PIN_59, COMCERTO_GPIO_63_32_PIN_OUTPUT);
+		msleep(1000);
+		//POWER_LED ON
+		writel(readl(COMCERTO_GPIO_63_32_PIN_OUTPUT) | GPIO_PIN_59, COMCERTO_GPIO_63_32_PIN_OUTPUT);
+		msleep(1000);
+	}
+
+	writel(readl(COMCERTO_GPIO_63_32_PIN_OUTPUT) | GPIO_PIN_59, COMCERTO_GPIO_63_32_PIN_OUTPUT);
+
+	remove_proc_entry("power_led",NULL);
+	
+	return 0;
+}
+
+void power_led_blink()
+{
+	proc_create("power_led",0,NULL,&power_led_fops);
+	
+        blink_thread = kthread_create(power_led_blink_thread,NULL,"power_led_blink");
+        if(blink_thread)
+        {
+                printk(KERN_INFO "%s: thread created ",__func__);
+                wake_up_process(blink_thread);
+        }
+}
